src/main/asciidoc/index.adoc

```diff
@@ -0,0 +1,426 @@
+= Vert.x for Groovy
+
+If you are using Maven or Gradle, add the following dependency to the _dependencies_ section of your
+project descriptor to access the Vert.x Core API and enable the Groovy support:
+
+* Maven (in your `pom.xml`):
+
+[source,xml,subs="+attributes"]
+----
+<dependency>
+ <groupId>io.vertx</groupId>
+ <artifactId>vertx-core</artifactId>
+ <version>${maven.version}</version>
+</dependency>
+<dependency>
+  <groupId>io.vertx</groupId>
+  <artifactId>vertx-lang-groovy</artifactId>
+  <version>${maven.version}</version>
+</dependency>
+----
+
+* Gradle (in your `build.gradle` file):
+
+[source,groovy,subs="+attributes"]
+----
+compile "io.vertx:vertx-core:${maven.version}"
+compile "io.vertx:vertx-lang-groovy:${maven.version}"
+----
+
+== Verticles
+
+=== Writing Verticles
+
+There are two alternatives to create Verticles in Groovy:
+
+a plain Groovy script
+a Groovy class implementing the {@link io.vertx.core.Verticle} interface or extending the {@link io.vertx.core.AbstractVerticle} class
+
+For example, the next snippet is a valid Groovy verticle:
+
+[source, groovy]
+----
+println "Hello from vertx"
+----
+
+On deployment, by default, Vert.x executes the script. Optionally, your script can provide the `startVertx` and
+`stopVertx` methods. Theses methods are called respectively when the verticle starts and stops:
+
+[source, groovy]
+----
+void vertxStart() {
+ println "starting"
+}
+
+void vertxStop() {
+ println "stopping"
+}
+----
+
+Alternatively, you can extend the {@link io.vertx.core.AbstractVerticle} class and implement the `start` and
+`stop` methods:
+
+[source, groovy]
+----
+import io.vertx.core.AbstractVerticle;
+
+class HelloWorldHttpVerticle extends AbstractVerticle {
+
+ void start() {
+  println("Starting")
+ }
+
+ void stop() {
+   println("Stopping")
+ }
+}
+----
+
+When Vert.x deploys the verticle it will call the `start` method, and when the method has completed the
+verticle will be considered started.
+
+You can also optionally override the `stop` method. This will be called by Vert.x when the verticle is undeployed
+and when the method has completed the verticle will be considered stopped.
+
+=== Accessing the vertx instance from a verticle
+
+Regardless the way you use to implement your verticle, you access the vert.x instance using the `vertx` variable /
+field.
+
+.Access to the vert.x instance in a Groovy script
+[source, groovy]
+----
+vertx.deployVerticle("another_verticle.groovy")
+----
+
+.Access to the vert.x instance in a Groovy class
+[source, groovy]
+----
+import io.vertx.lang.groovy.GroovyVerticle;
+
+class HelloWorldHttpVerticle extends GroovyVerticle {
```

Comment:
Isn't GroovyVerticle deprecated?

Reply:
I need to check

Reply:
It was deprecated in 3.x and removed in 4.0 according to #73 

So it should be replaced with AbstractVerticle